### PR TITLE
Protect reading external weights from up-dir

### PR DESCRIPTION
**Pull request by mbencer**
The base for this change is ONNX standard: https://github.com/onnx/onnx/blob/09a4e65bb098164491b021ffe563a559fbc1a808/onnx/external_data_helper.py#L188-L193

### CODE REVIEW

File: ngraph/frontend/onnx_import/src/utils/tensor_external_data.cpp [hunk 1]
```diff
@@ -47,8 +47,10 @@ namespace ngraph
             {
 #if defined(ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
                 std::wstring path = file_util::multi_byte_char_to_wstring(m_data_location.c_str());
+                file_util::sanitize_path(path, "\\.");
```
Comment (mbencer) [inline ngraph/frontend/onnx_import/src/utils/tensor_external_data.cpp:50]: It is implementation compatible with ONNX handling (`path.lstrip('/.')`), but I have a few doubts:
1. Correct path can be: /user/.onnx/weights.data -> current version won't handle it.
2. It doesn't block: ../weights.data case
Maybe better is: `sanitize_path(path, "..")`;?
or throwing exception if ".." is substring of the path? (with more user-friendly message)
@postrational What do you think?

Comment (tsocha) [reply ngraph/frontend/onnx_import/src/utils/tensor_external_data.cpp:50]: 2nd option is better IMO

Comment (mbencer) [reply ngraph/frontend/onnx_import/src/utils/tensor_external_data.cpp]: I've improved `sanitize_path` and move handling to `update_external_data_paths`.

**Review by tomdol — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Review by tsocha — APPROVED** [review]

File: ngraph/core/include/ngraph/file_util.hpp [hunk 2]
```diff
@@ -77,5 +77,12 @@ namespace ngraph
         /// \param str A null-terminated string
         /// \return A wide-char string
         NGRAPH_API std::wstring multi_byte_char_to_wstring(const char* str);
+
+        /// \brief Remove path components which would allow traversing up a directory tree
+        ///        from a base path.
+        /// \param path A path which will be santized
+        /// \return Path components to remove
+        NGRAPH_API std::string sanitize_path(const std::string& path,
```
Comment (postrational) [inline ngraph/core/include/ngraph/file_util.hpp:85]: `sanitize_path` should take only the `path` parameter. The other parameter is not a variable, we should define the sanitize logic entirely inside the function.

Also, we can give this function a more specific name, like `prevent_absolute_path` or something similar.

**Review by postrational — CHANGES_REQUESTED** [review]

File: ngraph/core/include/ngraph/file_util.hpp (hunk 2 above)
Comment (mbencer) [reply ngraph/core/include/ngraph/file_util.hpp]: - `to_remove` argument removed
- improved handling Windows-style cases
- I am not sure about `prevent_absolute_path` name. Paths like: ../../weights.data which are sanitized by this method are not absolute

**Review by postrational — APPROVED** [review]

**Pull request merged**